extensions/azdata/src/services/azdataToolService.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AzdataTool, IAzdataTool } from '../azdata';
+import * as constants from '../constants';
+
+
+export class AzdataToolService {
+	private _localAzdata: IAzdataTool | undefined;
+	constructor(private _memento: vscode.Memento) {
+		this._localAzdata = this._memento.get<AzdataTool>(constants.localAzdata);
```

Comment:
mm, I'm not sure about caching the tool like this. If we cache the tool and then the user uninstalls it that means any calls to the API will fail with an error about not finding the tool without also then prompting to reinstall, right?

I'd like this to be looked at separately so we can be sure we're not degrading any experiences here - I'd prefer you revert these changes for now and then follow up with a separate PR for just doing this stuff there.

Reply:
ok will remove this caching to the memento.

Reply:
reverted

Reply:
Just an FYI the code as it was on startup we were still updating the cache and using the tool discovered only in case of failure to discover where we continuing to use the tool from the cache.

Reply:
But I agree we should think through this functionality and bring it in a separate piece of work.